src/templates/varTemplates.ts

```diff
@@ -1,16 +1,38 @@
 import * as ts from 'typescript'
+import * as vsc from 'vscode'
+import * as _ from 'lodash'
+import { getVariableNameFromCallExpresion } from 'const-name'
 import { CompletionItemBuilder } from '../completionItemBuilder'
 import { BaseExpressionTemplate } from './baseTemplates'
+import { clipByLastIndex } from 'const-name/build/util'
 
 export class VarTemplate extends BaseExpressionTemplate {
   constructor(private keyword: 'var' | 'let' | 'const') {
     super(keyword)
   }
 
   buildCompletionItem(node: ts.Node, indentSize?: number) {
+    let varName: string
+
+    const deriveVariableName = vsc.workspace.getConfiguration('postfix', null).get<boolean>('deriveVariableName')
+    const expr = node.getText();
+    varName = deriveVariableName ?
+      expr.startsWith('new') ?
+        _.lowerFirst(
+          clipByLastIndex(/(.+?)\(/.exec(expr)[1], [' ', '.'])

```

Comment:
What does this line do? What's the problem it's trying to solve?

Reply:
This is actually internals of `const-name` lib itself: https://github.com/zardoy/const-name/blob/df3a021064b433aabc4eaa36b9d2d0ada96dd57f/src/index.ts#L46 I just pick everything before first `(`. See above, with using API we don't have to do this.